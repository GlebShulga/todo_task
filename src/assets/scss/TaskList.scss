.TaskList {
  width: 60%;
  text-align: center;
  padding-right: 0.5rem;
}

.TaskTable {
  display: flex;
  flex-direction: row;
  align-items: center;
  border: solid thin;
  padding: 0.5rem 0.5rem;
} // duplicated with category from CategoryList.scss

