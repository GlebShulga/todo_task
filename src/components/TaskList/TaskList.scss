.task-list {
  width: 45%;
  height: 100%;
  padding-right: 0.5rem;
}

.task-table {
  display: flex;
  flex-direction: row;
  align-items: center;
  border: solid thin;
  padding: 0.5rem 0.5rem;
}
