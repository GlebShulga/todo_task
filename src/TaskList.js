import React, { useEffect, useState } from "react";
import axios from "axios";
import editIcon from "./assets/images/editIcon.svg";
import "./assets/scss/TaskList.scss";

const TaskList = ({ choosenCategoryId }) => {
  const [taskList, setTaskList] = useState([]);

  useEffect(() => {
    axios("/api/v1/task")
      .then((res) => {
        const data = res.data;
        console.log(data, "data");
        setTaskList(data);
      })
      .catch((err) => console.log(err));
  }, []);

  // const statuses = ["new", "in progress", "done"];

  // const [activeStatus, setActiveStatus] = useState("all");

  return (
    <div className="TaskList">
      {taskList.map((task) => {
        return (
          choosenCategoryId === task.categoryId ?
          <div className="Task">
            <input
              type="checkbox"
              id="read_only"
              name="read_only"
              className=""
              // checked={data.read_only ?? false}
              // value={data.read_only ?? false}
              onChange={""}
            />
            <div key={task.taskId} className="Task-title">
              {task.title}
            </div>
            <img src={editIcon} className="Task-Edit_icon" alt="edit icon" />
          </div>
         : null
        );
      })}
    </div>
  );
};

export default TaskList;
