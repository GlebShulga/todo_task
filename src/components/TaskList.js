import React, { useEffect, useState } from "react";
import axios from "axios";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faCheckSquare,
} from "@fortawesome/free-solid-svg-icons";
import Task from './Task'
import EditTask from './EditTask'
import "../assets/scss/TaskList.scss";

const TaskList = ({ choosenCategory, isNewTaskCreated }) => {
  const [taskList, setTaskList] = useState([]);
  const [isEditingTaskMode, setEditingTaskMode] = useState(false);
  const [choosenTask, setChoosenTask] = useState({})

    const fetchTaskList = async () => {
      await axios("/api/v1/task")
        .then((res) => {
          const data = res.data;
          setTaskList(data);
        })
        .catch((err) => console.log(err));
    };

     const patchTask = async (taskId, status, title, description) => {
       await axios({
         method: "patch",
         url: "/api/v1/category",
         data: {
           taskId,
           status,
           title,
           description,
         },
       });
     };


  useEffect(() => {
    fetchTaskList()
  }, [isNewTaskCreated]);

  return (
    <div className="TaskList">
      {isEditingTaskMode && (
        <EditTask
          setEditingTaskMode={setEditingTaskMode}
          task={choosenTask}
          patchTask={patchTask}
          fetchTaskList={fetchTaskList}
        />
      )}
      {!isEditingTaskMode &&
        taskList.filter((task) => choosenCategory.categoryId === task.categoryId).map((task) => {
          return (
            <div className="TaskTable" key={task.taskId}>
              <FontAwesomeIcon
                icon={faCheckSquare}
                className={
                  task.status === 'done'
                    ? "checkMark"
                    : "checkMark_hidden"
                }
              />
              <Task
                task={task}
                setEditingTaskMode={setEditingTaskMode}
                setChoosenTask={setChoosenTask}
              />
            </div>
          );
        })}
    </div>
  );
};

export default TaskList;
