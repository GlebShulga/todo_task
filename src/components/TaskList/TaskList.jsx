import React, { useState, useEffect } from "react";
import { matchPath } from "react-router";
import { useLocation, useParams } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCheckSquare } from "@fortawesome/free-solid-svg-icons";
import Task from "../Task/Task";
import { useSelector } from "react-redux";
import "./TaskList.scss";

const TaskList = () => {
  const { pathname } = useLocation();
  const { category: choosenCategoryTitle } = useParams();
  const { taskList } = useSelector((s) => s.task);
  const { categoryList } = useSelector((s) => s.category);
  const [currentTaskList, setCurrentTaskList] = useState(taskList);

  const searchParams = matchPath(pathname, {
    path: "/:category/search/:subString",
  });

  const searchCriteria = searchParams?.params.subString;

  const chosenCategory = categoryList.find(
    (cat) => cat.categoryTitle === choosenCategoryTitle
  );

  useEffect(() => {
    if (searchCriteria) {
      setCurrentTaskList(
        taskList.filter((task) =>
          task.title.toLowerCase().includes(searchCriteria)
        )
      );
    } else {
      setCurrentTaskList(taskList);
    }
  }, [searchCriteria, taskList]);

  return (
    <div className="task-list">
      {currentTaskList
        .filter((task) => chosenCategory?.categoryId === task.categoryId)
        .map((task) => {
          return (
            <div className="task-table " key={task.taskId}>
              <FontAwesomeIcon
                icon={faCheckSquare}
                className={
                  task.status === "done" ? "check-mark" : "check-mark--hidden"
                }
              />
              <Task task={task} />
            </div>
          );
        })}
    </div>
  );
};

export default TaskList;
