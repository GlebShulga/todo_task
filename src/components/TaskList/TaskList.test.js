import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import TaskList from "./TaskList.jsx";

const middlewares = [thunk];
let mockStore;
const mockStoreConf = configureStore(middlewares);

describe("CategoryList Component", () => {
  beforeEach(() => {
    mockStore = mockStoreConf(
      {
        router: {
          location: { pathname: "/", search: "", hash: "", query: {} },
          action: "POP",
        },
        category: {
          categoryList: [
            {
              categoryId: "HD6CryOdGgYn4QJdKgs5G",
              parentCategoryId: "d1nN7oUmC1lTNtxGtEnbB",
              categoryTitle: "Nikita's tasks",
              lvl: 1,
            },
            {
              categoryId: "JO8Am3bmIIOMpnbYhwsVt",
              parentCategoryId: "5",
              categoryTitle: "category 2_1",
              lvl: 1,
            },
            {
              categoryId: "zi-jOQAPVMcHdpGIXd9jk",
              parentCategoryId: "zgcWQhZIDrbYzhSTGiOMC",
              categoryTitle: "test 7_3_1",
              lvl: 2,
            },
            {
              categoryId: "5ssRmvIUcEACosuj7UfbE",
              parentCategoryId: "Dxggo-RHZqSEoDByG1rB6",
              categoryTitle: "test 7_1_2",
              lvl: 2,
            },
            {
              categoryId: "zgcWQhZIDrbYzhSTGiOMC",
              parentCategoryId: "d1nN7oUmC1lTNtxGtEnbB",
              categoryTitle: "test 7_3",
              lvl: 1,
            },
            {
              categoryId: "24UUtTrRi-56blyyFBmuc",
              parentCategoryId: "d1nN7oUmC1lTNtxGtEnbB",
              categoryTitle: "test 7_2",
              lvl: 1,
            },
            {
              categoryId: "QbgbVXtxK2dLIHuQtlsne",
              parentCategoryId: "71QQpam8O1RZKItQjdVmY",
              categoryTitle: "test 7 1_1_2",
              lvl: 3,
            },
            {
              categoryId: "Anc96-W7mm7-gBn8_I7FZ",
              parentCategoryId: "Er20Gzc9qIcXW747oB56m",
              categoryTitle: "test 7_1_1_1_2",
              lvl: 4,
            },
            {
              categoryId: "EjscBBaQZwIpkEghrV8YG",
              parentCategoryId: "Er20Gzc9qIcXW747oB56m",
              categoryTitle: "test 7 1_1_1_1",
              lvl: 4,
            },
            {
              categoryId: "Er20Gzc9qIcXW747oB56m",
              parentCategoryId: "71QQpam8O1RZKItQjdVmY",
              categoryTitle: "test 7_1_1_1",
              lvl: 3,
            },
            {
              categoryId: "71QQpam8O1RZKItQjdVmY",
              parentCategoryId: "Dxggo-RHZqSEoDByG1rB6",
              categoryTitle: "test 7_1_1",
              lvl: 2,
            },
            {
              categoryId: "Dxggo-RHZqSEoDByG1rB6",
              parentCategoryId: "d1nN7oUmC1lTNtxGtEnbB",
              categoryTitle: "test 7_1",
              lvl: 1,
            },
            {
              categoryId: "d1nN7oUmC1lTNtxGtEnbB",
              parentCategoryId: null,
              categoryTitle: "test 7",
              lvl: 0,
            },
            {
              categoryId: "Wy1dXPxgnYdPJONBEFg5D",
              parentCategoryId: "wzvzJGZ_u7JggnKSlamOg",
              categoryTitle: "category 4_1_2",
              lvl: 2,
            },
            {
              categoryId: "wzvzJGZ_u7JggnKSlamOg",
              parentCategoryId: "RuQ3m0zrlNX9uY68zWCN1",
              categoryTitle: "test 4_1",
              lvl: 1,
            },
            {
              categoryId: "RuQ3m0zrlNX9uY68zWCN1",
              parentCategoryId: null,
              categoryTitle: "category 4",
              lvl: 0,
            },
            {
              categoryId: "6",
              categoryTitle: "category 2_3",
              parentCategoryId: "5",
              lvl: 1,
            },
            {
              categoryId: "8",
              categoryTitle: "category 2_2",
              parentCategoryId: "5",
              lvl: 1,
            },
            {
              categoryId: "7",
              categoryTitle: "category 2_1_1",
              parentCategoryId: "6",
              lvl: 2,
            },
            {
              categoryId: "5",
              categoryTitle: "category 2",
              parentCategoryId: null,
              lvl: 0,
            },
            {
              categoryId: "4",
              categoryTitle: "category 1_1_3",
              parentCategoryId: "2",
              lvl: 2,
            },
            {
              categoryId: "3",
              categoryTitle: "category 1_1_1",
              parentCategoryId: "2",
              lvl: 2,
            },
            {
              categoryId: "2",
              categoryTitle: "category 1_1",
              parentCategoryId: "1",
              lvl: 1,
            },
            {
              categoryId: "1",
              categoryTitle: "category 1",
              parentCategoryId: null,
              lvl: 0,
            },
            {
              categoryId: "PoMWPerFYXlI9_uCQH_Em",
              parentCategoryId: "Kbchx0i_c73rCkMJI8KCf",
              categoryTitle: "category 2_2_1_1",
              lvl: 3,
            },
            {
              categoryId: "Kbchx0i_c73rCkMJI8KCf",
              parentCategoryId: "8",
              categoryTitle: "category 2_2_1",
              lvl: 2,
            },
            {
              categoryId: "iQZpSMLpL0crGLqoS9M1d",
              parentCategoryId: null,
              categoryTitle: "category 3",
              lvl: 0,
            },
          ],
          rootCategoryList: [
            {
              categoryId: "d1nN7oUmC1lTNtxGtEnbB",
              parentCategoryId: null,
              categoryTitle: "test 7",
              lvl: 0,
            },
            {
              categoryId: "RuQ3m0zrlNX9uY68zWCN1",
              parentCategoryId: null,
              categoryTitle: "category 4",
              lvl: 0,
            },
            {
              categoryId: "5",
              categoryTitle: "category 2",
              parentCategoryId: null,
              lvl: 0,
            },
            {
              categoryId: "1",
              categoryTitle: "category 1",
              parentCategoryId: null,
              lvl: 0,
            },
            {
              categoryId: "iQZpSMLpL0crGLqoS9M1d",
              parentCategoryId: null,
              categoryTitle: "category 3",
              lvl: 0,
            },
          ],
          categoryTitleList: [
            "Nikita's tasks",
            "category 2_1",
            "test 7_3_1",
            "test 7_1_2",
            "test 7_3",
            "test 7_2",
            "test 7 1_1_2",
            "test 7_1_1_1_2",
            "test 7 1_1_1_1",
            "test 7_1_1_1",
            "test 7_1_1",
            "test 7_1",
            "test 7",
            "category 4_1_2",
            "test 4_1",
            "category 4",
            "category 2_3",
            "category 2_2",
            "category 2_1_1",
            "category 2",
            "category 1_1_3",
            "category 1_1_1",
            "category 1_1",
            "category 1",
            "category 2_2_1_1",
            "category 2_2_1",
            "category 3",
          ],
          chosenCategory: {},
          isFilterStatusDone: false,
          isCreateTaskModalOpen: false,
        },
        task: {
          taskList: [
            {
              categoryId: "3",
              taskId: "1",
              title: "Just do it",
              description: "bla",
              status: "new",
            },
            {
              categoryId: "3",
              taskId: "2",
              title: "Do it one more time",
              status: "new",
              description: "",
            },
            {
              categoryId: "4",
              taskId: "3",
              title: "Fly to the Earth",
              description: "bla-bla",
              status: "new",
            },
            {
              categoryId: "d1nN7oUmC1lTNtxGtEnbB",
              taskId: "pI2IO5LGgVTfjlWchzpKx",
              title: "test task",
              description: "",
              status: "new",
            },
            {
              categoryId: "HD6CryOdGgYn4QJdKgs5G",
              taskId: "m0hOnN18ldu69axREN0Tw",
              title: "fry big fish",
              status: "done",
              description: "FRY THE fish.",
            },
            {
              categoryId: "RuQ3m0zrlNX9uY68zWCN1",
              taskId: "454zuMqEHrrcDxwj1B5_p",
              title: "One more task",
              description: "",
              status: "done",
            },
            {
              categoryId: "7",
              taskId: "6Ni_9cmf7oHxsrxz-R2EP",
              title: "Second real task",
              description: "",
              status: "done",
            },
            {
              categoryId: "5",
              taskId: "Osau79o53nO1OAPtFmBXg",
              title: "Save the world today",
              description: "Save all species",
              status: "done",
            },
            {
              categoryId: "4",
              taskId: "5",
              title: "Fly to the Neptune",
              description: "I believe I can fly",
              status: "done",
            },
            {
              categoryId: "d1nN7oUmC1lTNtxGtEnbB",
              taskId: "mmYWc9uO3xPriRWkB5nGy",
              title: "Test task 1.0",
              description: "Test",
              status: "done",
            },
            {
              categoryId: "1",
              taskId: "BHD3mf0sWR5HRmtELuECx",
              title: "Edited task",
              description: "Task Limited Edition 5.0",
              status: "done",
            },
            {
              categoryId: "4",
              taskId: "4",
              title: "Fly to the Mars",
              description: "bla-bla-bla",
              status: "done",
            },
            {
              categoryId: "d1nN7oUmC1lTNtxGtEnbB",
              taskId: "9ZU08mIPEV4ZXjQKCanqY",
              title: "to-do",
              status: "done",
            },
            {
              categoryId: "6",
              taskId: "vz3GkbGCTfPzTXx2Shfmd",
              title: "task 2_1",
              status: "done",
              description: "Very important task",
            },
          ],
          chosenTask: {},
          searchCriteria: "",
          isEditingTaskMode: false,
          newCategoryIdForTask: "",
        },
      },
      {
        type: "GET_TASK_LIST",
        taskList: [
          {
            categoryId: "3",
            taskId: "1",
            title: "Just do it",
            description: "bla",
            status: "new",
          },
          {
            categoryId: "3",
            taskId: "2",
            title: "Do it one more time",
            status: "new",
            description: "",
          },
          {
            categoryId: "4",
            taskId: "3",
            title: "Fly to the Earth",
            description: "bla-bla",
            status: "new",
          },
          {
            categoryId: "d1nN7oUmC1lTNtxGtEnbB",
            taskId: "pI2IO5LGgVTfjlWchzpKx",
            title: "test task",
            description: "",
            status: "new",
          },
          {
            categoryId: "HD6CryOdGgYn4QJdKgs5G",
            taskId: "m0hOnN18ldu69axREN0Tw",
            title: "fry big fish",
            status: "done",
            description: "FRY THE fish.",
          },
          {
            categoryId: "RuQ3m0zrlNX9uY68zWCN1",
            taskId: "454zuMqEHrrcDxwj1B5_p",
            title: "One more task",
            description: "",
            status: "done",
          },
          {
            categoryId: "7",
            taskId: "6Ni_9cmf7oHxsrxz-R2EP",
            title: "Second real task",
            description: "",
            status: "done",
          },
          {
            categoryId: "5",
            taskId: "Osau79o53nO1OAPtFmBXg",
            title: "Save the world today",
            description: "Save all species",
            status: "done",
          },
          {
            categoryId: "4",
            taskId: "5",
            title: "Fly to the Neptune",
            description: "I believe I can fly",
            status: "done",
          },
          {
            categoryId: "d1nN7oUmC1lTNtxGtEnbB",
            taskId: "mmYWc9uO3xPriRWkB5nGy",
            title: "Test task 1.0",
            description: "Test",
            status: "done",
          },
          {
            categoryId: "1",
            taskId: "BHD3mf0sWR5HRmtELuECx",
            title: "Edited task",
            description: "Task Limited Edition 5.0",
            status: "done",
          },
          {
            categoryId: "4",
            taskId: "4",
            title: "Fly to the Mars",
            description: "bla-bla-bla",
            status: "done",
          },
          {
            categoryId: "d1nN7oUmC1lTNtxGtEnbB",
            taskId: "9ZU08mIPEV4ZXjQKCanqY",
            title: "to-do",
            status: "done",
          },
          {
            categoryId: "6",
            taskId: "vz3GkbGCTfPzTXx2Shfmd",
            title: "task 2_1",
            status: "done",
            description: "Very important task",
          },
        ],
      }
    );
  });

  it("TaskList snapshot", () => {
    const component = renderer.create(
      <Router>
        <Provider store={mockStore}>
          <TaskList />
        </Provider>
      </Router>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
