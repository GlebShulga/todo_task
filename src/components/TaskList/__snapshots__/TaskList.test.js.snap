// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList Component TaskList snapshot 1`] = `
<div
  className="task-list"
/>
`;
